src/auth/auth.controller.ts

```diff
@@ -75,4 +75,77 @@ export class AuthController {
       changePasswordRequest.newPassword,
     );
   }
+
+  @HttpCode(201)
+  @ApiOperation({
+    summary:
+      'Creates QR code for user to scan it for auth app (like google Authenticator)',
+  })
+  @UseGuards(JwtAuthGuard)
+  @Post('create-qr-2fa')
+  async createQrcodeFor2FA(
+    @UserId() userId: number,
+  ): Promise<CreateQrcodeFor2FA> {
+    const qrcode = await this.authService.createQrcodeFor2FA(userId);
+
+    return CreateQrcodeFor2FA.from(qrcode);
```

Comment:
hmm ok, for learning purposes we can leave the logic like this 👍🏻

If it was a production app I would leave the QR code generation in the frontend app hands, and here I would return a generated secret (some flows allow you to manually type secret to authenticator app instead of scanning a QR code)